java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/Server.java

```diff
@@ -185,6 +236,15 @@ public WorkspaceService getWorkspaceService() {
         @Override
         public void connect(LanguageClient client) {
             this.client = client;
```

Comment:
Wouldn't it be simpler if the type of `client` field was changed to `NbCodeLanguageClient`?

Reply:
If the capabilities are negotiated ?

Reply:
All the time. We can have the rich interface `NbCodeLanguageClient`, but only send messages that are allowed according to the capabilities.